api/data_explorer/util/elasticsearch_util.py

```diff
@@ -215,13 +251,44 @@ def get_elasticsearch_facet(es, elasticsearch_field_name, field_type):
             field=elasticsearch_field_name,
             interval=_get_bucket_interval(field_range))
 
-    # Handle sample facets in a special way since they are nested objects.
-    if elasticsearch_field_name.startswith('samples.'):
-        es_facet = NestedFacet('samples', es_facet)
-
+    nested_facet = _get_nested_facet(elasticsearch_field_name, es_facet,
+                                     nested_paths)
+    if nested_facet:
+        es_facet = nested_facet
     return es_facet
 
 
+def _get_nested_facet(elasticsearch_field_name, es_facet, nested_paths):
+    """
+    Returns a NestedFacet for the Elasticsearch field, if the field is nested.
+    """
+    parent = elasticsearch_field_name.rsplit('.', 1)[0]
+    nested_facet = None
+    # Traverse up the nesting levels from the leaf field, till we reach the root.
```

Comment:
Can you add somewhere:

Note there can be multiple levels of NestedFacet, eg NestedFacet(outer, NestedFacet(inner, es_facet))

Reply:
nit not fixed